woocommerce/woocommerce
# Add product name and checkboxes for Product details

### All Submissions:






### Changes proposed in this Pull Request:


This PR adds a couple of components to the Product details section.
The added components are:
- Name (TextControl)
- Feature this product (CheckboxControl + EnrichedLabel)


![screenshot-clean local-2022 08 11-08_30_17](https://user-images.githubusercontent.com/1314156/184124234-952c8603-aba5-40b3-b241-c80fad99d3e4.png)

Closes 28-gh-woocommerce/mothra-private and 36-gh-woocommerce/mothra-private.

### How to test the changes in this Pull Request:

1. Checkout this branch.
2. Verify that the texts look correct.
3. Verify that the help button (next to the checkbox text) and dialog work as expected. It should open on hover and track the event `wcadmin_add_product_learn_more` after pressing `Learn more`.
4. Verify that the event `wcadmin_woocommerce-add-product__checkbox-featured` is recorded after toggling the checkbox.
5. Verify that the input and checkbox values are being saved correctly.

### Other information:




### FOR PR REVIEWER ONLY:
